flutter/news/lib/features/frontpage/data/repositories/articles_repository.dart

```diff
@@ -18,9 +19,10 @@ class ArticlesRepository {
 
   Future<Result<List<Article>>> topHeadlines() async {
```

Comment:
On the design of this repository: 

I think here same good Command & Query Separation would give us more flexibility. 
A typical scenario is
```
**Given**  that the local cache has content
**When** the user want to refresh the content of the repository
**And** the request fails
**Then** show a snackbar
```

With this design the error won't be propagated to the consumer. 
I always find it more useful to differentiate `state retrieval` (Query) from `side effect generation` (Command) 

For example: 
```dart
class ArticleRepository {
    // You will always get the headlines from the local data source
    Stream topHeadlines() => localDataSource.getTopHeadlines()

   // When the command fails, the consumer will get an error 
   // When the command succeed the consumer will get the new headlines from the method above
  Future<void> fetchTopHeadlines() => remoteDataSource.fetchTopHeadlines()
     .then(localDataSource.save) // 
     .ignore();
}
```

Reply:
Yep, you are right, good idea for a follow up PR, lets keep this one however focused on introducing freezed, with no further changes

Reply:
Will try to separate the concerns  (command , query) on a follow up PR